.github/workflows/pull-request.yml

```diff
@@ -57,8 +59,41 @@ jobs:
           make html VERBOSE=1 SOF_DOC_BUILD=_build_doxy
           du -shc _build*/*
 
-  lax:
+      - name: prepare file for deploy
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/publish' }}
+        run: ./.github/actions/create-publish-folder.sh
+
+      # store the build result to artifact, used for later deploy or download for debug
+      # https://docs.github.com/en/actions/guides/storing-workflow-data-as-artifacts
+      - name: upload HTML for deploy
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/publish' }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: html
+          path: _build/html
+
+  deploy:
+    needs: build-basic
+    runs-on: ubuntu-20.04
```

Comment:
Btw, I'm assuming 20.04 since it's an LTS ?

Reply:
I don't think the next one is available on github yet.

Reply:
@lgirdwood @marc-hb GitHub Actions support `ubuntu-latest` also, but now the latest is still Ubuntu 20.04 (YES, it is a LTS)
We think stick to a known version will avoid some dependency issue later.